etc/errors/anvil.json

```diff
@@ -956,6 +956,230 @@
+                  "For example, the transaction may explicitly revert if the gas value provided is an even number.",
+                  "Therefore, the transaction failing with maximum gas may indicate a case of this exotic behavior.",
+                  "However, such behaviors make estimation impossible."
+                ]
+                ,
+                "likely_causes": [
+                  {
+                    "cause": "The transaction always halts.",
+                    "fixes": [
+                      "Inspect the source code of your contracts and debug the transaction to eliminate the cause of halt."
+                    ]
+                  },
+                  {
+                    "cause": "The transaction is malformed.",
+                    "fixes": [
+                      "Check the transaction for nonce mismatches, signature issues, and possibly paymaster-related errors.",
+                      "Do not provide explicit nonce in your RPC request if you don't need it -- otherwise you risk triggering nonce mismatch."
+                    ]
+                  }
+                ]
+
+              }
+            },
+            {
+              "name": "TransactionAlwaysReverts",
+              "code": 21,
+              "message": "Execution reverted during the gas estimation:\n{inner}",
+              "fields": [
+                {
+                  "name": "inner",
+                  "type": "Revert"
+                }
+              ],
+              "doc": {
+                "summary": "Transaction is not executable because it reverts with maximal allowed gas possible.",
+                "description": [
+                  "Before estimating gas, anvil-zksync first runs the transaction with maximum gas possible.",
+                  "This error occurs when this initial run results in a revert, suggesting that no amount of gas will make this transaction executable.",
+                  "",
+                  "To estimate the cost, anvil-zksync looks for a gas value starting from which the transaction succeeds.",
+                  "This works well if the transaction fails for all lower gas values and succeeds for all higher gas values.",
+                  "However, it is possible to craft transactions with exotic dependencies on gas.",
+                  "For example, the transaction may explicitly revert if the gas value provided is an even number.",
+                  "Therefore, the transaction failing with maximum gas may indicate a case of this exotic behavior.",
+                  "However, such behaviors make estimation impossible."
+                ],
+                "likely_causes": [
+                  {
+                    "cause": "The transaction always reverts, so it can't be estimated.",
+                    "fixes": [
+                      "Inspect the source code of your contracts and debug the transaction to eliminate the cause of revert."
+                    ]
```

Comment:
Additional one: "If transaction is expected to revert (e.g. for testing purposes), avoid estimating gas for it and provide gas limit manually"